src/savegame_json.cpp

```diff
@@ -1447,6 +1447,15 @@ void vehicle::deserialize(JsonIn &jsin)
 
     data.read("parts", parts);
 
+    // we persist the pivot anchor so that if the rules for finding
+    // the pivot change, existing vehicles do not shift around.
+    // Loading vehicles that predate the pivot logic is a special
+    // case of this, they will load with an anchor of (0,0) which
+    // is what they're expecting.
+    data.read("pivot", pivot_anchor[0]);
+    pivot_anchor[1] = pivot_anchor[0];
+    pivot_rotation[1] = pivot_rotation[0] = fdir;
```

Comment:
Won't those rules result in the vehicle shifting (possibly quite far) the first time it moves?
May be tolerable if there is no better way to solve it, but there may be some simple formula of normalizing the position against the new pivot.